crates/oxc_transformer/src/react/jsx.rs

```diff
@@ -706,11 +706,8 @@ impl<'a> ReactJsx<'a> {
             JSXElementName::Identifier(ident) => {
                 if ident.name == "this" {
                     self.ast().expression_this(ident.span)
-                } else if ident.name.chars().next().is_some_and(|c| c.is_ascii_lowercase()) {
-                    self.ast().expression_string_literal(ident.span, &ident.name)
                 } else {
```

Comment:
The last remaining thing is `JSXElementName::ThisExpression`, See https://github.com/oxc-project/oxc/pull/5223#issuecomment-2320957991 😀